docs/src/content/docs/modeling/reactions.md

```diff
@@ -0,0 +1,135 @@
+
+// React to firstName changes
+config.addReaction("firstName", (author) => {
+  // Business logic here
+  console.log(`Author name changed to ${author.firstName}`);
+});
+```
+## Named Reactions
+
+For debugging purposes, you can give reactions explicit names:
+
+```typescript
+config.addReaction(
+  "syncPublisherData",  // name for debugging
+  { publisher: ["name", "address"] },
+  (author) => {
+    // Business logic here
+  }
+);
+```
+
+The name will appear in error messages and logs, making it easier to trace which reaction is executing or causing issues.
+
+## Run-Once Reactions
+
+By default, reactions can run multiple times during a flush as the reactivity graph settles. If you need a reaction to run only once per flush, use the `runOnce` option:
+
+```typescript
+config.addReaction(
+  { runOnce: true },
+  "firstName",
+  (author) => {
+    // This will only run once per flush, even if firstName changes multiple times
+    sendNotification(author);
+  }
+);
+```
+
+You can also combine `runOnce` with a name:
+
+```typescript
+config.addReaction(
+  { name: "sendWelcomeEmail", runOnce: true },
+  ["firstName", "email"],
+  (author) => {
+    // Named and runs only once
+    queueWelcomeEmail(author);
+  }
+);
+```
+
+## Accessing Context
+
+Reactions receive the same context parameter as hooks, allowing access to the `EntityManager` and any custom context:
+
+```typescript
+config.addReaction("status", (author, ctx) => {
+  // Access the entity manager
+  const em = ctx.em;
+  // Access custom context (if configured)
+  await ctx.makeApiCall("author-status-changed");
+});
+```
+
+## Read-Only Relations
+
+If you want to pre-load relations in your reaction but don't want changes to those relations to trigger the reaction, you can mark them as read-only using the `:ro` suffix:.  This is not necessary for fields, as reactions are passed a `Loaded<T, H>` rather than a `Reacted<T, H>` so all primitive fields are available to read.
+
+```typescript
+config.addReaction(
+  { books_ro: ["title"] },
+  (author) => {
+    // This reaction triggers on book title changes, but not when books are added or removed from the underlying 
+    // relation.  The books relation, however, is still loaded and available to read.
+    const publishedBooks = author.books.get.filter(b => b.status === "published");
+  }
+);
+```
+
+## Best Practices
+
+1. **Keep reactions focused**: Each reaction should handle a single concern
+2. **Make reactions idempotent**: Since they can run multiple times, ensure they produce the same result
+3. **Avoid circular dependencies**: Don't create reactions where A triggers B which triggers A
+4. **Use read-only relations**: Mark relations as `:ro` when you only need to read them, not react to them
+5. **Consider `runOnce`**: If your reaction has side effects (like sending notifications), use `runOnce: true`
+6. **Name your reactions**: For easier debugging, especially in complex domain models
```

Comment:
I like the names, but not sure we need to promote them to as best practice; if they were "that strictly best practice" I think we'd require them them all the time? So maybe just delete this point.